### PR TITLE
Adding redux in react components

### PR DESCRIPTION
**Pull request by NATASHA-ct**
- [ ] Made sure the state remained immutable in reducers!

- [ ] Changed the default state in the books reducer from an empty array to an array with a few books. Displayed them in  React components.

- [ ] Import the Redux Provider and store in the main component of React app.

- [ ] Connected your app to the Redux store.

- [ ] Made sure that list of books displays books from the Redux store.

- [ ] Used useSelector to consume the state.

- [ ]  ✅ Added functionality for "Add new" button click: in React component responsible for adding new books(Form.js) 

- [ ] Imported Redux useDispatch.

- [ ] Set the data inputs in the local React state (set title and author )

- [ ] Dispatched a corresponding action (import Action Creator I created before ).

- [ ] ✅   Added functionality for the "Remove" button click: React component responsible for removing books

- [ ] Import Redux useDispatch.

- [ ] Dispatched a corresponding action (import Action Creator I created before ).

- [ ]  ✅  Added functionality for the "Remove" button click: React component responsible for categories,

- [ ] Import Redux useDispatch.

- [ ] Dispatched a corresponding action (import Action Creator that I created before ).

### CODE REVIEW

File: src/components/Category.js [hunk 1]
```diff
@@ -1,7 +1,18 @@
...
+  const checkHandler = () => {
+    dispatch(checkStatus());
+  };
+  return (
+    <div>
+      <button type="button" onClick={checkHandler}>
+        Check status
+      </button>
+    </div>
+  );
```
Comment (Nemwel-Boniface) [inline src/components/Category.js]: - [ ] Great job so far! :clap: . I have realized that when I click this button nothing gets displayed (The "Under Construction"). Kindly include this to meet all the project requirements.

image

**Review by Nemwel-Boniface — CHANGES_REQUESTED** [review]
## Hello @NATASHA-ct            :wave: , 
I'm Nemwel your reviewer for this PR :man_beard: 

You have done an amazing job with your Bookstore project so far :clap: 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️  


## ♻️ Required Changes ♻️

_Check the comments under the review._

## [Optional] suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding! :partying_face: 👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Lordkaito — APPROVED** [review]
Hi @NATASHA-ct, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by BertrandConxy — APPROVED** [review]
Hi @NATASHA-ct , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 💪 
- No linter errors
- project requirements are met
- project works fine
- good PR description
- correct flow is used

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @BertrandConxy in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**